src/Cli/dotnet/commands/dotnet-test/TestApplication.cs

```diff
@@ -50,9 +55,104 @@ public async Task<int> RunAsync(bool enableHelp)
                 RedirectStandardError = true
             };
 
-            return await StartProcess(processStartInfo);
+            //
+            _namedPipeConnectionLoop = Task.Run(async () => await WaitConnectionAsync(_cancellationToken.Token));
+            var result = await StartProcess(processStartInfo);
+
+            _namedPipeConnectionLoop.Wait();
+            return result;
+        }
+
+        private async Task WaitConnectionAsync(CancellationToken token)
+        {
+            try
+            {
+                _pipeConnection = new(_pipeNameDescription, OnRequest, NamedPipeServerStream.MaxAllowedServerInstances, token, skipUnknownMessages: true);
+                _pipeConnection.RegisterAllSerializers();
+
+                await _pipeConnection.WaitConnectionAsync(token);
+            }
+            catch (Exception ex)
```

Comment:
Shall we fail in case of cancelled token?

Reply:
I added some code to avoid failing in case of cancelled token.